src/go/rdctl/pkg/reg/reg.go

```diff
@@ -0,0 +1,160 @@
+//	boolean: true if the current entry is empty
+//	error
+func convertToRegFormat(pathParts []string, v reflect.Value, jsonTag string) ([]string, bool, error) {
+	kind := v.Kind()
+	switch kind {
+	case reflect.Struct:
+		numFields := v.NumField()
+		scalarReturnedLines := make([]string, 0, numFields)
+		nestedReturnedLines := []string{}
+		for i := 0; i < numFields; i++ {
+			fieldTag := v.Type().Field(i).Tag.Get("json")
+			fieldName := strings.Replace(fieldTag, ",omitempty", "", 1)
+			newRetLines, isEmpty, err := convertToRegFormat(append(pathParts, fieldName), v.Field(i), fieldName)
+			if err != nil {
+				return nil, true, err
+			}
+			if isEmpty {
+				continue
+			}
+			if len(newRetLines[0]) == 0 {
+				nestedReturnedLines = append(nestedReturnedLines, newRetLines...)
+			} else {
+				scalarReturnedLines = append(scalarReturnedLines, newRetLines...)
+			}
+		}
+		if len(scalarReturnedLines) == 0 && len(nestedReturnedLines) == 0 {
+			return nil, true, nil
+		}
+		retLines := []string{"", fmt.Sprintf("[%s]", strings.Join(pathParts, "\\"))}
+		retLines = append(retLines, scalarReturnedLines...)
+		retLines = append(retLines, nestedReturnedLines...)
+		return retLines, false, nil
+	case reflect.Ptr:
+		if v.IsNil() {
+			return []string{}, true, nil
+		} else {
+			return convertToRegFormat(pathParts, v.Elem(), jsonTag)
+		}
+	case reflect.Slice, reflect.Array:
+		// Currently, all arrays in the options are arrays of strings
+		numValues := v.Len()
+		if numValues == 0 {
+			return nil, true, nil
+		}
+		arrayValues := make([]string, numValues)
+		for i := 0; i < numValues; i++ {
+			arrayValues[i] = v.Index(i).String()
+		}
+		return []string{fmt.Sprintf(`"%s"=hex(7):%s`, jsonTag, stringToMultiStringHexBytes(arrayValues))}, false, nil
+	case reflect.Map:
+		numValues := len(v.MapKeys())
+		if numValues == 0 {
+			return nil, true, nil
+		}
+		retLines := []string{"", fmt.Sprintf("[%s]", strings.Join(pathParts, "\\"))}
+		for _, key := range v.MapKeys() {
+			keyAsString := key.String()
+			innerLines, isEmpty, err := convertToRegFormat(append(pathParts, keyAsString), v.MapIndex(key), keyAsString)
+			if err != nil {
+				return nil, isEmpty, err
+			} else if !isEmpty {
+				retLines = append(retLines, innerLines...)
+			}
+		}
+		return retLines, false, nil
+	case reflect.Interface:
+		if v.IsNil() {
+			return nil, true, nil
+		}
+		return convertToRegFormat(pathParts, v.Elem(), jsonTag)
+	case reflect.Bool:
+		boolValue := map[bool]int{true: 1, false: 0}[v.Bool()]
+		return []string{fmt.Sprintf(`"%s"=dword:%d`, jsonTag, boolValue)}, false, nil
+	case reflect.Int, reflect.Int8, reflect.Int16,
+		reflect.Int32, reflect.Uint, reflect.Uint8, reflect.Uint16,
+		reflect.Uint32:
+		return []string{fmt.Sprintf(`"%s"=dword:%x`, jsonTag, v.Int())}, false, nil
+	case reflect.Int64, reflect.Uint64:
+		return []string{fmt.Sprintf(`"%s"=qword:%x`, jsonTag, v.Int())}, false, nil
+	case reflect.Float32, reflect.Float64:
```

Comment:
I think floats are normally stored as `REG_BINARY`? Not sure we have any right now anyway though.

Reply:
Except out maps aren't fully typed. If I take out that handler, given this code:
```
		jsonBody := `{
 "WSL": {
   "integrations": {
			"fish": true,
			"sheep": false,
			"cows": 17,
			"owls": "stuff"
		}
  }
}`
		header := "HKEY_CURRENT_USER\\SOFTWARE\\Policies\\Rancher Desktop\\defaults"
		lines, err := JsonToReg("hkcu", "defaults", jsonBody)
		assert.NoError(t, err)
```

I'll get this error when testing:
```
	convertToRegFormat: don't know how to process kind: float64, (reflect.Value), value: 17 for var cows
```